src/picknik_ur_base_config/description/picknik_ur_macro.xacro

```diff
@@ -0,0 +1,89 @@
+        <xacro:include filename="$(find picknik_ur_base_config)/description/pinch_links.xacro"/>
+
+        <!-- Initial positions for simulations (Mock Hardware and Gazebo) -->
+        <xacro:property name="initial_positions_file" default="$(arg initial_positions_file)"/>
+        <xacro:property name="simulation" default="$(arg simulation)"/>
+        <xacro:property name="visual_parameters_file" default="$(arg visual_parameters_file)"/>
+        <xacro:if value="${simulation == 'gazebo'}">
+            <xacro:property name="use_fake_hardware" default="false"/>
+        </xacro:if>
+        <xacro:if value="${not simulation == 'gazebo'}">
+            <xacro:property name="use_fake_hardware" default="$(arg use_fake_hardware)"/>
+        </xacro:if>
+
+        <link name="${parent}"/>
+        <!-- arm -->
+        <xacro:ur_robot
+                name="$(arg name)"
+                tf_prefix=""
+                parent="${parent}"
+                joint_limits_parameters_file="$(arg joint_limits_parameters_file)"
+                kinematics_parameters_file="$(arg kinematics_parameters_file)"
+                physical_parameters_file="$(arg physical_parameters_file)"
+                visual_parameters_file="$(arg visual_parameters_file)"
+                use_fake_hardware="${use_fake_hardware}"
+                initial_positions="${xacro.load_yaml(initial_positions_file)}"
+                sim_ignition="${simulation == 'gazebo'}"
+                headless_mode="$(arg headless_mode)"
+                robot_ip="$(arg robot_ip)"
+                use_tool_communication="$(arg use_tool_communication)"
+                tool_voltage="$(arg tool_voltage)"
+                tool_device_name="$(arg tool_device_name)"
+                script_filename="$(find ur_robot_driver)/resources/ros_control.urscript"
+                output_recipe_filename="$(find ur_robot_driver)/resources/rtde_output_recipe.txt"
+                input_recipe_filename="$(find ur_robot_driver)/resources/rtde_input_recipe.txt">
+            <origin xyz="0 0 0" rpy="0 0 0"/>
+        </xacro:ur_robot>
+
+        <xacro:unless value="$(arg has_tool_changer)">
+            <xacro:property name="child" default="$(arg child)"/>
+            <xacro:if value="${not child == 'tool0'}">
+                <link name="${child}"/>
+                <joint name="${child}_joint" type="fixed">
+                    <parent link="tool0"/>
+                    <child link="${child}"/>
+                    <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
+                </joint>
+            </xacro:if>
+        </xacro:unless>
+
+        <!-- pinch geometry for moveit motion planning. see https://github.com/UniversalRobots/Universal_Robots_ROS2_Driver/issues/611 -->
+        <xacro:if value="$(arg use_pinch_links)">
+            <xacro:forearm_pinch_link/>
+            <xacro:wrist_3_pinch_link/>
+        </xacro:if>
```

Comment:
Should these be part of the attachments macro as well? At the least the `tool0` link definition should be, right?

Reply:
Actually, tool0 is the link defined by the UR robot xacro, so we have no control over that. It is possible to put the tool changer in the attachments though if that makes sense.

Reply:
Sorry, didn't realize `tool0` was the parent, not the child. I meant the child link definition. Pinch link you could argue could be a top level thing since it's an extra addition for motion planning to be well behaved.